dtfs-central-api/src/repositories/tfm-deals-repo/tfm-deal-cancellation.repo.ts

```diff
@@ -167,6 +173,20 @@ export class TfmDealCancellationRepo {
       throw new DealNotFoundError(dealId.toString());
     }
 
-    return { cancelledDealUkefId: dealId };
+    const facilityCollection = await this.getFacilityCollection();
+
+    await facilityCollection.updateMany(
+      { 'facilitySnapshot.dealId': { $eq: new ObjectId(dealId) } },
+      flatten({
+        'tfm.facilityStage': TFM_FACILITY_STAGE.RISK_EXPIRED,
+        auditRecord: generateAuditDatabaseRecordFromAuditDetails(auditDetails),
+      }),
+    );
+
+    const updatedFacilities = await facilityCollection.find({ 'facilitySnapshot.dealId': { $eq: new ObjectId(dealId) } }).toArray();
+
```

Comment:
I haven't worked with MongoDB for a while - is it possible to return the facilities from `.updateMany` instead of having to do a `.find` afterwards?

Reply:
Unfortunately it seems like you can't - there's `findOneAndUpdate` which allows you to update a single facility and return the updated facility in the response, but as soon as you update multiple facilities at a time you can't return them all 😞